### PR TITLE
Optimize the publishing

### PR DESCRIPTION
**Pull request by sbrunner**
Be able to publish many images at one time e.g.: x, x.y, x.y.z and latest.
Run Snyk only one time.

See also: https://github.com/camptocamp/c2cgeoportal/pull/10301

The idea is:
- Add an argument to specify all the images we want to export
- Add an argument to specify the tag we would like to use for Snyk
=> publish all the images
=> scan only one image => one time

### CODE REVIEW

File: c2cciutils/scripts/publish.py [hunk 1]
```diff
@@ -66,6 +66,8 @@ def main() -> None:
     parser = argparse.ArgumentParser(description="Publish the project.")
     parser.add_argument("--group", default="default", help="The publishing group")
     parser.add_argument("--version", help="The version to publish to")
+    parser.add_argument("--docker-versions", help="The version to publish to Docker registry")
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:69]:
```suggestion
    parser.add_argument("--docker-versions", help="The versions to publish on Docker registry, comma separated, ex: 'x,x.y,x.y.z,latest'.")
```

File: c2cciutils/scripts/publish.py [hunk 2]
```diff
@@ -205,54 +207,69 @@ def main() -> None:
...
+        images_snyk: Set[str] = set()
         for image_conf in docker_config.get("images", []):
-            if (
-                image_conf.get("group", c2cciutils.configuration.PUBLISH_DOCKER_IMAGE_GROUP_DEFAULT)
-                == args.group
-            ):
-                for tag_config in image_conf.get(
-                    "tags", c2cciutils.configuration.PUBLISH_DOCKER_IMAGE_TAGS_DEFAULT
+            versions = args.docker_versions.split(",") if args.docker_versions else [version]
+            for docker_version in versions:
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:213]: Is it not possible to move this for loop just around the publish action (c2cciutils.publish.docker).
Do we need multiple calendar events ?
As I see it should be possible and simpler.

Comment (sbrunner) [reply c2cciutils/scripts/publish.py]: Looks good, thanks :-)

Comment (sbrunner) [reply c2cciutils/scripts/publish.py]: Done :-)

File: c2cciutils/scripts/publish.py [hunk 3]
```diff
@@ -205,7 +210,9 @@ def main() -> None:
 
         images_src: Set[str] = set()
         images_full: List[str] = []
+        images_snyk: Set[str] = set()
         for image_conf in docker_config.get("images", []):
+            versions = args.docker_versions.split(",") if args.docker_versions else [version]
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:215]: Seems it can be be outside for loop

**Review by arnaud-morvan — APPROVED** [review]

**Pull request merged**